internal/venice-client-common/src/main/java/com/linkedin/venice/stats/SplitOtlpHttpMetricExporter.java

```diff
@@ -0,0 +1,146 @@
+package com.linkedin.venice.stats;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.common.export.MemoryMode;
+import io.opentelemetry.sdk.metrics.Aggregation;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * A MetricExporter that splits the input metrics into batches and exports each batch separately.
+ * maxBatchSize is the maximum number of metric points allowed in a single batch.
+ *
+ * For instance, the below metric has metric points and if maxBatchSize is 1, then this will be
+ * split into 2 batches:
+ * ....
+ * name=venice.thin_client.call_time, description=Latency based on all responses, unit=MILLISECOND,
+ * type=EXPONENTIAL_HISTOGRAM, data=ImmutableExponentialHistogramData{aggregationTemporality=DELTA,
+ * points=[
+ * ImmutableExponentialHistogramPointData{......},
+ * ImmutableExponentialHistogramPointData{......}
+ * ]}},
+ * ....
+ */
+public class SplitOtlpHttpMetricExporter implements MetricExporter {
+  private final MetricExporter delegate;
+  private final int maxBatchSize;
+
+  public SplitOtlpHttpMetricExporter(MetricExporter delegate, int maxBatchSize) {
+    this.delegate = delegate;
+    this.maxBatchSize = maxBatchSize;
+  }
+
+  /**
+   * Export the input metrics in batches by leveraging the delegate exporter.
+   */
+  @Override
+  public CompletableResultCode export(Collection<MetricData> metrics) {
+    Collection<CompletableResultCode> results = new ArrayList<>();
+    splitIntoBatches(metrics, maxBatchSize, results);
```

Comment:
If we could have splitIntoBatches returns `Collection<CompletableResultCode>`, export would be a bit simpler.